### PR TITLE
Main kaggle comp

### PR DESCRIPTION
**Pull request by RasmusOrsoe**
A bit of polishing required to make a graphnet example on the kaggle dataset.

### CODE REVIEW

**Comment by RasmusOrsoe** [comment]
@asogaard Where did all the checks go?

File: src/graphnet/data/dataset.py [hunk 1]
```diff
@@ -598,36 +598,51 @@ def _create_graph(
 
         # Additionally add original features as (static) attributes
         for index, feature in enumerate(graph.features):
-            graph[feature] = graph.x[:, index].detach()
```
Comment (asogaard) [inline src/graphnet/data/dataset.py:601]: Ahh, this is because the Kaggle-dataset feature `x` conflicts with the torch-geometric built-in feature `x` referring to the features tensor?

File: src/graphnet/data/dataset.py [hunk 2]
```diff
@@ -598,36 +598,51 @@ def _create_graph(
...
+            if feature not in ["x"]:
+                graph[feature] = graph.x[:, index].detach()
 
         # Add custom labels to the graph
         for key, fn in self._label_fns.items():
             graph[key] = fn(graph)
-
         return graph
 
     def _get_labels(self, truth_dict: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (asogaard) [inline src/graphnet/data/dataset.py:609]: We really ought to reconsider this function, and same for the double-bang one...

File: src/graphnet/models/detector/icecube.py [hunk 3]
```diff
@@ -43,6 +43,34 @@ def _forward(self, data: Data) -> Data:
         return data
 
 
+class IceCube86Kaggle(Detector):
```
Comment (asogaard) [inline src/graphnet/models/detector/icecube.py:46]: Does the Kaggle data represent the main array only or also DeepCore?

**Review by asogaard — APPROVED** [review]
Looks really good! Thanks, @RasmusOrsoe.

File: src/graphnet/data/dataset.py (hunk 1 above)
Comment (RasmusOrsoe) [reply src/graphnet/data/dataset.py:601]: Yes.

File: src/graphnet/models/detector/icecube.py (hunk 3 above)
Comment (RasmusOrsoe) [reply src/graphnet/models/detector/icecube.py:46]: Both!

File: src/graphnet/data/dataset.py (hunk 2 above)
Comment (RasmusOrsoe) [reply src/graphnet/data/dataset.py:609]: Yes, I agree. Maybe in the future, we could make a collection of `Labels` that the `Dataset` just defaults to.

File: src/graphnet/models/detector/icecube.py (hunk 3 above)
Comment (asogaard) [reply src/graphnet/models/detector/icecube.py:46]: Then I figure it should be named `IceCubeDeepCoreKaggle` or just `IceCubeKaggle`.

Comment (RasmusOrsoe) [reply src/graphnet/models/detector/icecube.py]: It has been renamed to `IceCubeKaggle`
